### PR TITLE
Add language config mappings from VSC extensions

### PR DESCRIPTION
**Pull request by 4086606**
The [icon pack](https://github.com/PKief/vscode-material-icon-theme) doesn't include extensions that can instead be associated with a reconfigurable language ID.

Let's pull in mutually exclusive language configuration data from Microsoft repos so that this data can be used to compute and apply icons accurately.

### CODE REVIEW

**Comment by shivapoudel** [comment]
@4086606 Just like we copied the `material-icons.json` into `iconMap.json` is there any repo we can look for to generate that language-related JSON it is because we don't wanna store those files here manually!

**Comment by 4086606** [comment]
👀 Don't hardcode those please!

It's compiled from every contributes.languages in the Microsoft organisation:

* Search for "contributes languages filename:package.json org:microsoft"
* Download the files to one place
* Scan for /([ \t]+)"languages": \[\n(.*\n)+?\1\]/
* Delete any folders with "sample" in the name
* Convert into lang-config map
* Move any file extensions not prefixed with . into file names *
* Remove anything that's mapped in material-icons.json *
* Do some remapping *
** properties extensions ^git -> git
** properties extensions ==editorconfig -> editorconfig
** properties extensions ^npm -> npm
** properties extensions -> settings
** properties names -> git
** vb -> visualstudio
** tablegen -> fsharp (custom)
** python -> python-misc (custom)
** shellscript -> console
** juliamarkdown -> julia (custom)
** git-commit -> git
** git-rebase -> git
** ignore -> vscode (custom)
* Do some merging:
** perl6 -> perl
** jsonc -> json
* Do some deletions: *
** Delete hlsl
** Delete log extensions
** Delete ignore extensions
* Remove definitions present in material-icons.json *

(Anything with a * after needs empty arrays removed as a post-completion step.)

**Comment by 4086606** [comment]
@Claudiohbsantos ideally this PR would precede the work in #7 and get merged first

**Comment by 4086606** [comment]
@shivapoudel I made the package able to generate the new config file automatically in 4da4285

**Comment by 4086606** [comment]
I also switched away from the camelcase naming convention because this repo ships with JSON files

**Comment by Claudiohbsantos** [comment]
Thanks for the work on this. I'm travelling for a quick break until mid next week. I'll make sure to review this asap when I'm back

File: scripts/extract-svg-html.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    fs.readdir(iconsPath)
+      .then((icons) => {
+        console.log(`[${opts.task}:2/4] Fetch map of icon file names and SVG data.`);
+        return Promise.all(
+          icons.map((name) => fs.readFile(path.resolve(iconsPath, name), { encoding: 'utf8' }))
+        ).then((xmlContent) => xmlContent.map((xml, i) => [icons[i], parse(xml)]))
+      })
+      .then((svgEntries) => {
+        console.log(`[${opts.task}:3/4] Determine HTML code and viewbox for icon cache.`);
+        svgEntries.map((entry) => [
```
Comment (Claudiohbsantos) [inline scripts/extract-svg-html.js:20]: We're missing a return here, without which the `build-src` npm script fails

```suggestion
        return svgEntries.map((entry) => [
```

**Review by Claudiohbsantos — CHANGES_REQUESTED** [review]
This looks great, thanks! I ran it locally and tested on chrome without a problem! 

There is just a small issue on a missing return that is preventing `build` from running but once that's included I can approve :)

A small non-critical request: do you have the link to a repository in which the languages added can be seen as an example when compared to the current `master` version? If you don't have one handy you were using for testing don't worry about it

**Comment by 4086606** [comment]
I would be ideal to run the build scripts and publish all that to a `dev` branch. I'll have that done in an hour.

**Comment by 4086606** [comment]
@Claudiohbsantos done, see https://github.com/4086606/github-material-icons-extension/tree/dev

**Review by Claudiohbsantos — APPROVED** [review]

**Comment by Claudiohbsantos** [comment]
Thanks for the changes. I'll merge this and then jump onto the other PR asap.

I'm not sure I understand the advantage of checking in generated files into git. Could you elaborate on how that would be better than having devs run the build scripts locally?

**Comment by 4086606** [comment]
That's a good idea. Could automate it all as a self updating repo via cron job

**Comment by 4086606** [comment]
That's what Phillip does over at the main repo. It's honestly super useful if you can't run the build scripts or download Chromium.

**Comment by Claudiohbsantos** [comment]
@4086606 That makes sense. With the latest PR just merged downloading Chromium shouldn't be necessary anymore, which I believe will alleviate most of this issue. For now I'll hold off on checking in the generated files, just because I generally find more issues than benefits to having to wrangle them while navigating branches and their potentially different versions, and because I don't believe this project will have/need as much contribution as the VSCode extension, so I think keeping the repo simpler makes things easier for now. I appreciate the suggestion though, and will keep it in mind for the future 👍

**Pull request merged**